components/idenpub/idenpub.go

```diff
@@ -0,0 +1,184 @@
+package idenpub
+
+import (
+	"bytes"
+	"math"
+	"sync"
+
+	"github.com/iden3/go-iden3-core/db"
+	"github.com/iden3/go-iden3-core/merkletree"
+)
+
+var (
+	cacheIKey  = []byte("cacheI")
+	cltRootKey = []byte("cltRoot")
+	rotRootKey = []byte("rotRoot")
+	retRootKey = []byte("retRoot")
+	rotKey     = []byte("rot")
+	retKey     = []byte("ret")
+)
+
+// IdenPub is a interface, that for the moment will be satisfied at least by IdenPubHTTP & IdenPubIPFS.
+type IdenPub interface {
+	Publish()
+}
+
+// IdenPubHTTP satisfies the IdenPub interface, and stores in a leveldb the published RoT & ReT to be returned when requested.
+type IdenPubHTTP struct {
+	sync.RWMutex
```

Comment:
This exposes the RWMutex functions publicly for the IdenPubHTTP type.  Is this the desired outcome?